sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/OtlpConfigUtil.java

```diff
@@ -114,6 +115,22 @@ static void configureOtlpExporterBuilder(
     }
   }
 
+  static void configureOtlpAggregationTemporality(
+      ConfigProperties config, Consumer<AggregationTemporality> setAggregationTemporality) {
+    String temporalityStr = config.getString("otel.exporter.otlp.metrics.temporality");
+    if (temporalityStr == null) {
+      return;
+    }
+    AggregationTemporality temporality;
+    try {
+      temporality = AggregationTemporality.valueOf(temporalityStr.toUpperCase());
+    } catch (IllegalArgumentException e) {
+      throw new ConfigurationException(
+          "Unrecognized aggregation temporality: " + temporalityStr, e);
+    }
+    setAggregationTemporality.accept(temporality);
```

Comment:
Move this and the declaration of `temporality` var into the `try` block.

Reply:
I don't think that's right. The try block is trying to explicitly catch a distinct type of error that occurs when parsing a string to an enum. Including this in the try block would cause errors in `setAggregationTemporality#accept`  to manifest with misleading exceptions that read `Unrecognized aggregation temporality:`.

There's no expectation that an error will occur on this line, and unit tests exist which exercise it according to its usage so I think its appropriate to keep it outside a try / catch.

Reply:
I guess I just think it's kinda ugly to declare a scoped variable that is like half initialized and forces the reader to visually scan to grok that you won't hit the setter in the invalid case. I dunno. 

What I've done elsewhere is to make a similar `valueOf` but called `fromString` or something that encapsulates the try/catch behavior. Then the caller looks a lot cleaner...but the `ConfigurationException` is NOT specific to the enum and wouldn't make sense in this case, so I dunno.

Reply:
Yeah enum parsing has always been frustratingly verbose.